### PR TITLE
file-sorter: fix fake resolved ts event breaks CRTs restraint

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix bug found in https://github.com/pingcap/ticdc/issues/754

### What is changed and how it works?

Use `CRTs-1` as fake resolved ts, where `CRTs` is commit ts of a real event. The root cause is sorted events could have the same commit ts, use `CRTs-1` as resolved ts is safe and eliminates this problem.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

File: cdc/puller/file_sorter.go [hunk 1]
```diff
@@ -415,7 +415,7 @@ func (fs *FileSorter) rotate(ctx context.Context, resolvedTs uint64) error {
 			// file sorter outputs all events in this rotate round.
 			rowCount += 1
 			if rowCount%defaultAutoResolvedRows == 0 {
-				fs.output(ctx, model.NewResolvedPolymorphicEvent(item.entry.CRTs))
+				fs.output(ctx, model.NewResolvedPolymorphicEvent(item.entry.CRTs-1))
```
Comment (leoppro) [inline cdc/puller/file_sorter.go:418]: what happened if the current `item.entry.CRTs-1` is equal to the `item.entry.CRTs` of the previous event

Comment (amyangfei) [reply cdc/puller/file_sorter.go:418]: Events are sorted by commit-ts here, and the previous is guaranteed to be sent to output chan

Comment (overvenus) [inline cdc/puller/file_sorter.go:418]: Please add comments about `-1`.

Comment (amyangfei) [reply cdc/puller/file_sorter.go:421]: addressed

**Pull request merged**